torch/nn/functional.py

```diff
@@ -3737,6 +3737,42 @@ def triplet_margin_loss(anchor, positive, negative, margin=1.0, p=2, eps=1e-6, s
                                      swap, reduction_enum)
 
 
+def triplet_margin_loss_with_distance(anchor, positive, negative, distance_function=None, is_similarity_function=False,
+                                      margin=1.0, swap=False, reduction="mean"):
+    # type: (Tensor, Tensor, Tensor, Optional[Callable[[Tensor, Tensor], Tensor]], bool, float, bool, str) -> Tensor
+    r"""
+    See :class:`~torch.nn.TripletMarginLossWithDistance` for details
+    """
+    if torch.jit.is_scripting():
+        raise NotImplementedError("F.triplet_margin_loss_with_distance does not support JIT. "
```

Comment:
Can you elaborate on why this doesn't supporting being scripted?

Reply:
Does this need code like triplet_margin_loss has to `handle_torch_function`?

Reply:
Sounds good. Would you add.a comment explaining the lack of scripting support. Does it work with tracing?

Reply:
> Does it work with tracing?

I'm able to do the following, but I'm not sure if this qualifies as "working with tracing"

```
>>> import torch.nn.functional as F
>>> F.triplet_margin_loss_with_distance
<function triplet_margin_loss_with_distance at 0x7feb97ad21f0>
>>> import torch
>>> anch = torch.randn(5, 5)
>>> pos = torch.randn(5, 5)
>>> neg = torch.randn(5, 5)
>>> torch.jit.trace(F.triplet_margin_loss_with_distance, (anch, pos, neg))
<torch.jit.ScriptFunction object at 0x7febac7f70e0>
```